cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java

```diff
@@ -81,6 +84,15 @@ public void validate()
                 throw new RuntimeException("KeyStore is empty");
             }
 
+            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();)
+            {
+                Certificate cert = keyStore.getCertificate(aliases.nextElement());
+                if (cert instanceof X509Certificate)
+                {
+                    ((X509Certificate) cert).checkValidity();
+                }
+            }
```

Comment:
Can we collapse the for lops in lines 87 and 96:
```suggestion
            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();)
            {
                String alias = aliases.nextElement();
                Certificate cert = keyStore.getCertificate(alias);
                if (cert instanceof X509Certificate)
                {
                    ((X509Certificate) cert).checkValidity();
                }

                Key key = keyStore.getKey(alias, password);
                if (key != null && key instanceof PrivateKey)
                {
                    return;  // KeyStore contains a private key
                }
            }
```

Reply:
ah, I just realized that the second loop looks for a PK< and it if finds it, it exits the validation as successful. Please disregard this comment.